### PR TITLE
Spring mapper dto2

### PR DESCRIPTION
**Pull request by jah701**

### CODE REVIEW

File: src/main/java/com/dev/cinema/model/dto/UserResponseDto.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.dev.cinema.model.dto;
+
+import lombok.Data;
+
+@Data
+public class UserResponseDto {
+    private Long id;
+    private String name;
```
Comment (AnaLanda) [inline src/main/java/com/dev/cinema/model/dto/UserResponseDto.java:8]: Maybe we'll need the user's email here as well)

**Review by AnaLanda — COMMENTED** [review]
Looks good 😃

File: src/main/java/com/dev/cinema/model/dto/UserResponseDto.java (hunk 1 above)
Comment (jah701) [reply src/main/java/com/dev/cinema/model/dto/UserResponseDto.java:8]: I'll let our user decide if he whether wants to hide email or not as long as it's required to login ;) 
But youre right, id and name only is less 🤔

Comment (jah701) [reply src/main/java/com/dev/cinema/model/dto/UserResponseDto.java:8]: Actually there is nothing else i can add in this case 🤔

File: src/main/java/com/dev/cinema/controller/CinemaHallController.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    @GetMapping
+    public List<CinemaHallResponseDto> getAll() {
+        return cinemaHallService.getAll().stream()
+                .map(cinemaHallMapper::cinemaHallToDto)
+                .collect(Collectors.toList());
+    }
+
+    @PostMapping
+    public CinemaHall add(@RequestBody CinemaHallRequestDto dto) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controller/CinemaHallController.java]: let's not return entity here

either return void, or ResponseDto

File: src/main/java/com/dev/cinema/controller/OrderController.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        this.userService = userService;
+        this.shoppingCartService = shoppingCartService;
+        this.orderMapper = orderMapper;
+    }
+
+    @PostMapping("/complete")
+    public Order completeOrder(@RequestParam Long userId) {
+        ShoppingCart shoppingCart = shoppingCartService.getByUser(userService.getById(userId));
+        return orderService.completeOrder(shoppingCart.getTickets(),
+                userService.getById(userId));
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controller/OrderController.java]: let's save User in a variable

and again, do not return the model itself

File: src/main/java/com/dev/cinema/controller/ShoppingCartController.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    public ShoppingCartResponseDto getByUser(@RequestParam Long userId) {
+        return shoppingCartMapper.cartToDto(
+                shoppingCartService.getByUser(userService.getById(userId)));
+    }
+
+    @PostMapping("/movie-sessions")
+    public void addMovieSessionToSoppingCart(@RequestParam Long userId,
+                                                     @RequestParam Long sessionId) {
+        shoppingCartService.addSession(movieSessionService.getById(sessionId),
+                userService.getById(userId));
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controller/ShoppingCartController.java]: local variables make the code more readable

File: src/main/java/com/dev/cinema/controller/ShoppingCartController.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    @GetMapping("/by-user")
+    public ShoppingCartResponseDto getByUser(@RequestParam Long userId) {
+        return shoppingCartMapper.cartToDto(
+                shoppingCartService.getByUser(userService.getById(userId)));
+    }
+
+    @PostMapping("/movie-sessions")
+    public void addMovieSessionToSoppingCart(@RequestParam Long userId,
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controller/ShoppingCartController.java]:
```suggestion
    public void addMovieSessionToShoppingCart(@RequestParam Long userId,
```

File: src/main/java/com/dev/cinema/model/dto/OrderRequestDto.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package com.dev.cinema.model.dto;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import lombok.Data;
+
+@Data
+public class OrderRequestDto {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/dto/OrderRequestDto.java]: do you use this?

File: src/main/java/com/dev/cinema/model/dto/OrderResponseDto.java [hunk 7]
```diff
@@ -0,0 +1,12 @@
+package com.dev.cinema.model.dto;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import lombok.Data;
+
+@Data
+public class OrderResponseDto {
+    private Long userId;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/dto/OrderResponseDto.java:9]: let's maybe add orderId as well?

File: src/main/java/com/dev/cinema/model/dto/ShoppingCartRequestDto.java [hunk 8]
```diff
@@ -0,0 +1,10 @@
+package com.dev.cinema.model.dto;
+
+import java.util.List;
+import lombok.Data;
+
+@Data
+public class ShoppingCartRequestDto {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/dto/ShoppingCartRequestDto.java]: do you use this?

File: src/main/java/com/dev/cinema/model/dto/TicketResponseDto.java [hunk 9]
```diff
@@ -0,0 +1,10 @@
+package com.dev.cinema.model.dto;
+
+import lombok.Data;
+
+@Data
+public class TicketResponseDto {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/dto/TicketResponseDto.java]: I think we can remove this Dto and use `List<Long> ticketIds` field in ShoppingCartResponseDto and OrderResponseDto instead

File: src/main/java/com/dev/cinema/service/impl/ShoppingCartServiceImpl.java [hunk 10]
```diff
@@ -40,6 +40,7 @@ public ShoppingCart getByUser(User user) {
     public void registerNewShoppingCart(User user) {
         ShoppingCart shoppingCart = new ShoppingCart();
         shoppingCart.setUser(user);
+        shoppingCart.setId(user.getId());
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/impl/ShoppingCartServiceImpl.java]: isn't it set automatically with @MapsId annotation?

File: src/main/resources/db.properties [hunk 11]
```diff
@@ -6,4 +6,4 @@ db.password=p81gORq76
 
 #Hibernate properties
 hibernate.show_sql=true
-hibernate.hbm2ddl.auto=create-drop
+hibernate.hbm2ddl.auto=update
```
Comment (KseniiaMakarova) [inline src/main/resources/db.properties]: why have you decided to change this?

File: src/main/java/com/dev/cinema/service/mapper/UserMapper.java [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+
+import com.dev.cinema.model.User;
+import com.dev.cinema.model.dto.UserResponseDto;
+import org.springframework.stereotype.Component;
+
+@Component
+public class UserMapper {
+    public UserResponseDto userToDto(User user) {
+        UserResponseDto dto = new UserResponseDto();
+        dto.setName(user.getName());
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/mapper/UserMapper.java:11]: are all ResponseDto fields set?

File: src/main/java/com/dev/cinema/service/mapper/CinemaHallMapper.java [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+import com.dev.cinema.model.dto.CinemaHallResponseDto;
+import org.springframework.stereotype.Component;
+
+@Component
+public class CinemaHallMapper {
+    public CinemaHallResponseDto cinemaHallToDto(CinemaHall cinemaHall) {
+        CinemaHallResponseDto dto = new CinemaHallResponseDto();
+        dto.setCapacity(cinemaHall.getCapacity());
+        dto.setDescription(cinemaHall.getDescription());
+        return dto;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/mapper/CinemaHallMapper.java:15]: same

File: src/main/java/com/dev/cinema/service/mapper/TicketMapper.java [hunk 14]
```diff
@@ -0,0 +1,25 @@
+package com.dev.cinema.service.mapper;
+
+import com.dev.cinema.model.Ticket;
+import com.dev.cinema.model.dto.MovieSessionResponseDto;
+import com.dev.cinema.model.dto.TicketResponseDto;
+import org.springframework.stereotype.Component;
+
+@Component
+public class TicketMapper {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/mapper/TicketMapper.java]: I think we can remove this

we will return ticketIds
if necessary, the additional info about tickets can be fetched given these ids

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/controller/CinemaHallController.java (hunk 2 above)
Comment (jah701) [reply src/main/java/com/dev/cinema/controller/CinemaHallController.java]: Agree :)

File: src/main/java/com/dev/cinema/controller/ShoppingCartController.java (hunk 5 above)
Comment (jah701) [reply src/main/java/com/dev/cinema/controller/ShoppingCartController.java]: :D

File: src/main/resources/db.properties (hunk 11 above)
Comment (jah701) [reply src/main/resources/db.properties]: I only need it for testing

File: src/main/java/com/dev/cinema/service/impl/ShoppingCartServiceImpl.java (hunk 10 above)
Comment (jah701) [reply src/main/java/com/dev/cinema/service/impl/ShoppingCartServiceImpl.java]: Sorry forgot to clear

File: src/main/java/com/dev/cinema/service/impl/UserServiceImpl.java [hunk 15]
```diff
@@ -26,7 +25,12 @@ public User add(User user) {
...
     @Override
-    public Optional<User> findByEmail(String email) {
-        return userDao.findByEmail(email);
+    public User getById(Long id) {
+        return userDao.getById(id).get();
+    }
+
+    @Override
+    public User findByEmail(String email) {
+        return userDao.findByEmail(email).get();
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/impl/UserServiceImpl.java]: I would say that it is better to still return Optional here (and call optional.get() in UserController)

the problem is that, if user enters wrong email, previously the AuthenticationExeption was thrown from AuthService - and now NoSuchElementException (you call `get()` on empty Optional in UserServiceImpl)

File: src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java [hunk 16]
```diff
@@ -22,9 +23,10 @@ public AuthenticationServiceImpl(ShoppingCartService shoppingCartService,
 
     @Override
     public User login(String email, String password) throws AuthenticationException {
-        Optional<User> userFromDb = userService.findByEmail(email);
-        if (userFromDb.isPresent() && isPasswordValid(password, userFromDb.get())) {
-            return userFromDb.get();
+        log.info("[LOGIN] User with email " + email + " is trying to login. . .");
+        User userFromDb = userService.findByEmail(email);
+        if (userFromDb != null && isPasswordValid(password, userFromDb)) {
+            return userFromDb;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java]: so let's maybe not change the code in this class

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/service/impl/UserServiceImpl.java (hunk 15 above)
Comment (jah701) [reply src/main/java/com/dev/cinema/service/impl/UserServiceImpl.java]: Ok, ive only changed it according to our qna :)

Comment (KseniiaMakarova) [reply src/main/java/com/dev/cinema/service/impl/UserServiceImpl.java]: sometimes one may not notice some issues from the first glance)

Comment (jah701) [reply src/main/java/com/dev/cinema/service/impl/UserServiceImpl.java]: Youre right, ofcourse )

**Comment by jah701** [comment]
Сори, вроде запускал чекстайл 🐒

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**